modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -32,9 +32,13 @@ func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpart
 
 	msg := channeltypes.NewMsgChannelOpenInit(portID, icatypes.VersionPrefix, channeltypes.ORDERED, []string{connectionID}, icatypes.PortID, icatypes.ModuleName)
 	handler := k.msgRouter.Handler(msg)
-	if _, err := handler(ctx, msg); err != nil {
+
+	res, err := handler(ctx, msg)
+	if err != nil {
 		return err
 	}
 
+	ctx.EventManager().EmitEvents(res.GetEvents())
```

Comment:
we should add a comment explaining why this is necessary so it isn't deleted later